ml-package-versions.yml

```diff
@@ -0,0 +1,177 @@
+      "== 1.5.0": ["torchvision==0.6.0", "scikit-learn"]
+      "== 1.5.1": ["torchvision==0.6.1", "scikit-learn"]
+      "== 1.6.0": ["torchvision==0.7.0", "scikit-learn"]
+      "> 1.6.0": ["torchvision", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+  autologging:
+    minimum: "1.4.0"
+    maximum: "1.7.0"
+    requirements:
+      "== 1.4.0": ["torchvision==0.5.0", "pytorch-lightning", "scikit-learn"]
+      "== 1.5.0": ["torchvision==0.6.0", "pytorch-lightning", "scikit-learn"]
+      "== 1.5.1": ["torchvision==0.6.1", "pytorch-lightning", "scikit-learn"]
+      "== 1.6.0": ["torchvision==0.7.0", "pytorch-lightning", "scikit-learn"]
+      "> 1.6.0": ["torchvision", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+tensorflow-1.x:
+  package_info:
+    pip_release: "tensorflow"
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["h5py<3.0"]
+    run: |
+      pytest tests/tensorflow/test_tensorflow_model_export.py --large
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["h5py<3.0"]
+    run: |
+      pytest tests/tensorflow_autolog/test_tensorflow_autolog.py --large
+
+tensorflow-2.x:
+  package_info:
+    pip_release: "tensorflow"
+    install_dev: |
+      pip install tf-nightly
+
+  models:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    run: |
+      pytest tests/tensorflow/test_tensorflow2_model_export.py --large
+
+  autologging:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    requirements:
+      "< 2.2": ["h5py<3.0"]
+    run: |
+      pytest --verbose tests/tensorflow_autolog/test_tensorflow2_autolog.py --large
+
+keras-tf1.x:
+  package_info:
+    pip_release: "tensorflow"
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["keras==2.2.5", "h5py<3.0", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
+    run: |
+      pytest tests/keras/test_keras_model_export.py --large
+
+  autologging:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["keras==2.2.5", "h5py<3.0"]
+    run: |
+      pytest --verbose tests/keras_autolog/test_keras_autolog.py --large
+
+keras-tf2.x:
+  package_info:
+    pip_release: "keras"
+    install_dev: |
+      # In keras >= 2.4.0, `keras` simply redirects to `tensorflow.keras`:
+      # https://github.com/keras-team/keras#multi-backend-keras-and-tfkeras
+      pip install tf-nightly 'keras>=2.4.0'
+
+  models:
+    minimum: "2.3.1"
+    maximum: "2.4.3"
+    requirements:
+      "== 2.3.1":
+        ["tensorflow==2.2.1", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
+      "> 2.3.1": ["tensorflow", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
+    run: |
+      pytest tests/keras/test_keras_model_export.py --large
+
+  autologging:
+    minimum: "2.3.1"
+    maximum: "2.4.3"
+    requirements:
+      "== 2.3.1": ["tensorflow==2.2.1"]
+      "> 2.3.1": ["tensorflow"]
```

Comment:
Note that `> 2.3.1` includes the dev version.

Reply:
I don't fully understand why we need to use the dev version here. From what I can tell, recent versions of Keras depend on TensorFlow, so presumably installing the correct version of Keras should pull in a compatible version of TensorFlow. For that reason, I think it's probably sufficient to just specify an unpinned `tensorflow` dependency. In fact, that only reason to specify `tensorflow` at all would be to handle the case where Keras drops TensorFlow as a dependency in the future, which seems somewhat unlikely.

Let me know if this reasoning makes sense.

Reply:
Edit: Misinterpreted this, doesn't seem like we're actually pulling in the dev version of TensorFlow. Can you clarify what you mean here @harupy ?

Reply:
Sorry for the cofusion. I just wanted to tell `"> x.x.x": [package1, package2]` means that package1 and package2 will be included in the dev test.

Reply:
Ah, I see. `> 2.3.1` includes the dev version of *keras*, not *TensorFlow*. Makes sense!